### ISSUE TITLE
Add point sources to the scene

### ISSUE THREAD

**Issue by jintonic** [jintonic/shine#10]
A point source is a point where radiation comes out. The user should be able to drag and drop a source point to the scene and fine tune its x, y, z value.

There should be a drop down list for the user to choose what kind of particles can be emitted from the point. The list of particles can be found here: <https://github.com/jintonic/gears/blob/master/tutorials/sources/gps.txt#L40> The default value is geantino

There should be a field for the user to specify the energy and a dropdown list to specify the unit of energy value, the list is here: <https://github.com/jintonic/gears/blob/master/tutorials/sources/gps.txt#L73>. The default value is keV

There should be a way to specify the range of solid angles within which the radiation exists. This is similar to the camera setup. We may be able to borrow the camera code to realize this. The idea is shown in the picture here: <https://publications.chitkara.edu.in/a-geant4-study-of-a-gamma-ray-collimation-array>

**Comment by jintonic** [comment]
The energy units are case-sensitive. "EV" is not correct. "eV" is. In general, we need to pay attention to the case of a word.

**Comment by bestdeveloper23** [comment]
Should I add two row for "source type" and "source shape".
And geometric size: "halfx, halfy, halfz, radius, inner_radius on right panel.

**Comment by bestdeveloper23** [comment]
I should reproduce uppercase letters as lowercase letters. right?

**Comment by jintonic** [comment]
No. EV should be eV. V should always be capitalized. Please type strictly at they shown on <https://github.com/jintonic/gears/blob/master/tutorials/sources/gps.txt#L73>

**Comment by jintonic** [comment]
4 rows:
row 1: source type
row 2: source shape
row 3: half x, y, z:
row 4: inner radius, outer radius

**Comment by bestdeveloper23** [comment]
I see.

**Comment by jintonic** [comment]
Could you please finish this work as well? I already paid this in advance.

**Comment by bestdeveloper23** [comment]
Honestly, I'm not sure if the results for this will be foolproof.
But I will do my best.

**Comment by jintonic** [comment]
Don't worry too much about it. We need the structure first. We'll make it better later.

**Comment by bestdeveloper23** [comment]
Okay

**Comment by bestdeveloper23** [comment]
In fact, while working on gdml, I lost the code that previously implemented the sources for other sources.
I already made the basic for it.
After making it, I asked you for an image.
So I should recover it.

**Comment by jintonic** [comment]
If everything is in github, you should be able to go back to history

**Comment by bestdeveloper23** [comment]
And since the bulk replace action was performed and then the discard operation was performed, it cannot be found.
But it's in my brain.
Do not worry.

**Comment by bestdeveloper23** [comment]
Let's change the name of the point source to Source and make it possible to specify the source type and shape in the properties panel on the right. 
And let's create an option to set the geometric dimensions for the corresponding source type and shape.

**Comment by jintonic** [comment]
That's a good idea! Could you please also change the icon? I prefer something cleaner, for example, this one: https://nohat.cc/f/radioactive-symbol-clipart-transparent-background-radiation-sign/m2i8i8m2H7K9K9b1-201907240059.html

**Comment by bestdeveloper23** [comment]
I completed right panel.
Pushed code to a new branch "dev".
Please check

**Comment by jintonic** [comment]
How can I check? The dev branch cannot be deployed. Shall I use vite to check locally?

**Comment by jintonic** [comment]
Shall I accept your push request?

**Comment by jintonic** [comment]
Why do you have to make a push request? You can directly work on this repo.

**Comment by bestdeveloper23** [comment]
Yes. I can.

**Comment by bestdeveloper23** [comment]
I merged myself.
You can check now.

**Comment by jintonic** [comment]
Thank you! Please fix the following on the right panel when source is selected:
- [ ] add Half Z, remove the last two "Half X".
- [ ] Hide the line of "Type PerspectiveCamera".
- [ ] Hide the line of "UUID" for source.

**Comment by bestdeveloper23** [comment]
I updated code.
Please check.
What's the next?
